File: src/main/java/com/server/ordering/repository/ReviewRepository.java
package com.server.ordering.repository;

import com.server.ordering.domain.Review;
import lombok.RequiredArgsConstructor;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.stereotype.Repository;

import javax.persistence.EntityManager;
import java.util.List;

@Repository
@RequiredArgsConstructor
public class ReviewRepository {

    private final EntityManager em;

    public void save(Review review) {
        em.persist(review);
    }

    public Review findOne(Long reviewId) {
        return em.find(Review.class, reviewId);
    }

    public List<Review> findAllWithOrderWithCustomerByRestaurantId(Long restaurantId) {
        return em.createQuery("select distinct m from Review m" +
                        " left join fetch m.order o" +
                        " left join fetch o.customer where m.restaurant.id =:restaurantId order by m.id desc ", Review.class)
                .setParameter("restaurantId", restaurantId)
                .getResultList();
    }

    public void remove(Review review) {
        em.remove(review);
    }
}

File: src/main/kotlin/com/byeoru/ordering_server/repository/ReviewRepository.kt
package com.byeoru.ordering_server.repository

import com.byeoru.ordering_server.domain.Review
import org.springframework.stereotype.Repository
import javax.persistence.EntityManager

@Repository
class ReviewRepository(private val em: EntityManager) {

    fun save(review: Review) = em.persist(review)

    fun findOne(reviewId: Long): Review = em.find(Review::class.java, reviewId)

    fun findAllWithOrderWithCustomerByRestaurantId(restaurantId: Long): List<Review> {
        return em.createQuery(
            "select distinct m from Review m" +
                    " left join fetch m.order o" +
                    " left join fetch o.customer where m.restaurant.id =:restaurantId" +
                    " order by m.id desc ", Review::class.java
        )
            .setParameter("restaurantId", restaurantId)
            .resultList
    }

    fun remove(reviewId: Long) {
        val review = em.find(Review::class.java, reviewId)
        em.remove(review)
    }
}